specs/airr-schema.yaml

```diff
@@ -2255,154 +2249,223 @@ Rearrangement:
-            description: CDR3 end position in the query sequence (1-based closed interval).
+            description: Junction region end position in the alignment (1-based closed interval).
+            x-airr:
+                miairr: false
+        junction:
+            type: string
+            description: >
+                Junction region nucleotide sequence, where the junction is defined as
+                the CDR3 plus the two flanking conserved codons.
+            x-airr:
+                miairr: false
+        junction_aa:
+            type: string
+            description: >
+                Amino acid translation of the junction.
+            x-airr:
+                miairr: false
         rearrangement_count:
             type: integer
-            description: Number of rearrangements represented by this clone
-        unique_rearrangement_count:
-            type: integer
-            description: Number of unique rearrangements represented by this clone
-        clone_count:
-            type: integer
-            description: Number of instances for this clone
+            description: Number of rearrangements included in this clone
+            x-airr:
+                miairr: false
         seed_id:
             type: string
             Description: rearrangment id for seed, null if no seed
-        mean_mut_freq:
-            type: number
-            description: Mean mutation frequency across sequences in the clone.
-        insertions:
-            type:  string
-            description: Clone insertion mutations
-        deletions:
-            type:  string
-            description: Clone deletion mutations
-
-# Clone information for a specific repertoire within a single clone processing.
-CloneRepertoire:
-    discriminator: AIRR
-    type: object
-    properties:
-        clone_id:
-            type: string
-            description: Identifier for the clone.
-        data_processing_id:
-            type: string
-            description: Identifier for the data processing object.
-        repertoire_id:
-            type: string
-            description: Identifier for the repertoire object.
+            x-airr:
+                miairr: false
 
 # 1-to-n relationship for a clone to its trees.
 Tree:
     discriminator: AIRR
     type: object
+    required:
+        - tree_id
+        - clone_id
+        - newick_string
     properties:
         tree_id:
             type: string
             description: Identifier for the tree.
+            x-airr:
+                miairr: false
+                required: true
         clone_id:
             type: string
             description: Identifier for the clone.
+            x-airr:
+                miairr: false
+                required: true
         newick_string:
             type: string
             description: Newick string of the tree edges.
+            x-airr:
+                miairr: false
+                required: true
         node_list:
```

Comment:
Should this be called `nodes` or `node_dict` or `nodes_by_seqid` or something like that to reflect its structure?

Reply:
I will change this now if no one objects. Any preference among the choices here or suggestions?

Reply:
I think it should be `node_id`, similar to how `Clone` has `sequence_id`

Reply:
From a consistency perspective this makes sense, but actually both cases kind of confuse me by using a singular name to refer to a structure containing many of those things. Is there a reason for doing it this way?

Reply:
They get the same name because they are referencing the same thing. The values in `Clone.sequence_id` are exactly those in `Rearrangement.sequence_id` and are used to cross-reference. Same here. OTOH, I could also see using `sequence_ids` or `sequence_id_list` in `Clone` if you find that more salutary. Check with @javh, I think it was originally his suggestion to use the same field name.

Reply:
Got it. I am in favor of using the same name for the same thing but unless there is a reason to have a list of `thing` have the exact same name as `thing` (e.g. programming/querying use case requiring them to have the same name) without some plural suffix, then I think a plural suffix like you suggest woud make sense.

Reply:
We have a `type` property, so we've been avoiding type identifying names (Hungarian notation). So just `nodes` (or `node`), similar to `sample` in `Repertoire`,  `sequencing_files` in `SequencingRun`, etc.

We don't have another dictionary-like object in the schema, but this looks correct to me according to:
https://swagger.io/docs/specification/data-models/dictionaries/
I'll have to defer to @schristley though.

And, yeah, I think `sequence_id` as the key is correct.

Reply:
The issue with `sequence_id` is it is only unique within a "file" and may not be persistent, so we should also allow `rearrangement_id` to be used.

Reply:
But `sequence_id` is required, while `rearrangement_id` is not...
I could see using `rearrangement_id` as a kind of future-proofing, though, since it will be important if identifying `Clone`s across multiple `Repertoire`s.